### ISSUE TITLE
error in dwipreproc

### ISSUE THREAD

**Issue by scottkolbe** [MRtrix3/mrtrix3#1014]
Hi there, 
Got an error during dwipreproc. I am using a read segmented EPI scheme so not sure if this is messing up the checking of the pe scheme. Here is everything from the terminal:

```
ubuntu@scott:~/ms_mri/ex_vivo/DWI_170603$ dwipreproc dwi_denoised.mif dwi_denoised_preprocessed.mif -pe_dir AP -rpe_pair -se_epi b0s.mif -readout_time 0.031
dwipreproc: 
dwipreproc: Note that this script makes use of commands / algorithms that have relevant articles for citation; INCLUDING FROM EXTERNAL SOFTWARE PACKAGES. Please consult the help page (-help option) for more information.
dwipreproc: 
dwipreproc: Generated temporary directory: /home/ubuntu/ms_mri/ex_vivo/DWI_170603/dwipreproc-tmp-IKO5QK/
Command:  mrconvert /home/ubuntu/ms_mri/ex_vivo/DWI_170603/dwi_denoised.mif /home/ubuntu/ms_mri/ex_vivo/DWI_170603/dwipreproc-tmp-IKO5QK/dwi.mif
Command:  mrconvert /home/ubuntu/ms_mri/ex_vivo/DWI_170603/b0s.mif /home/ubuntu/ms_mri/ex_vivo/DWI_170603/dwipreproc-tmp-IKO5QK/topup_in.mif
dwipreproc: Changing to temporary directory (/home/ubuntu/ms_mri/ex_vivo/DWI_170603/dwipreproc-tmp-IKO5QK/)
Traceback (most recent call last):
  File "/home/ubuntu/Applications/mrtrix3/bin/dwipreproc", line 280, in <module>
    if not scheme_times_match(dwi_pe_scheme, dwi_manual_pe_scheme):
  File "/home/ubuntu/Applications/mrtrix3/bin/dwipreproc", line 259, in scheme_times_match
    if abs(header_line[3] - manual_line[3]) > 5e-3:
NameError: global name 'header_line' is not defined
```

cheers 
Scott

**Comment by Lestropie** [comment]
No, nothing to do with readout-segmented, that's just me derping variable names at some point (had to re-write `dwipreproc` from scratch, was bound to bugger something up). Will only cause an error if manually providing the readout time. Will be fixed once the tests for #1015 pass (about a half hour).

**Comment by scottkolbe** [comment]
thanks Rob

**Comment by Lestropie** [comment]
P.S. @scottkolbe: The fact that that function is being called at all suggests that you've used an up-to-date `mrconvert`, which has captured the phase encoding scheme from the DICOM headers? If so, I'm curious to know whether or not the script gives a warning regarding a mismatch between what's stored in the header, and the total readout time you have provided manually. I *think* that what `mrconvert` imports should be correct for readout-segmented EPI; for phase-encode-segmented, I'd be less optimistic...

**Comment by scottkolbe** [comment]
It's true that the readout time I put in was rounded to the nearest ms. The header contains a read-out time to one more decimal of precision. this was not flagged by the script.

**Comment by Lestropie** [comment]
Yes, there's a bit of tolerance built in there. As long as they're close enough, it won't provide a warning. But hopefully that at least means that the total readout time imported from the DICOM header matches whatever information you used to independently determine the total readout time.

**Comment by scottkolbe** [comment]
it is accurate, yet rounded to a 10th of a millisecond which I think should be totally fine.

**Comment by lmlacerda** [comment]
Hi @Lestropie,

Hope you are well.

Instead of opening a new issue, I thought I would just report a minor error that I got while running dwipreproc: if by any chance the temp folder is saved within a path that contains a space, there is an error and the function won't continue. I tend to avoid using spaces, but got myself in a folder with spaces when debugging my code.

As an example I created a **"te st"** folder and you can see the following error message appears if I run my command from inside this folder:

dwipreproc: [03;32mGenerated temporary directory: /Users/luis/Desktop/Projects/Advanced_visual_system/**te st**/dwipreproc-tmp-9DT0RP/[0m
[03;36mCommand:[0m  mrconvert "/Users/luis/Desktop/Projects/Advanced_visual_system/Controls/109_multishell_pre_processed/109_multishell_denoised.mif" /Users/luis/Desktop/Projects/Advanced_visual_system/**te st**/dwipreproc-tmp-9DT0RP/dwi.mif

dwipreproc: [03;32m[0m
dwipreproc: [01;31m[ERROR] Command failed: mrconvert "/Users/luis/Desktop/Projects/Advanced_visual_system/Controls/109_multishell_pre_processed/109_multishell_denoised.mif" /Users/luis/Desktop/Projects/Advanced_visual_system/**te st**/dwipreproc-tmp-9DT0RP/dwi.mif[0m[03;34m (dwipreproc:136)[0m

dwipreproc: [03;32mOutput of failed command:[0m
**mrconvert: [ERROR] Expected exactly 2 arguments (3 supplied)**
mrconvert: [ERROR] Usage: mrconvert input output
mrconvert: [ERROR] Yours: mrconvert /Users/luis/Desktop/Projects/Advanced_visual_system/Controls/109_multishell_pre_processed/109_multishell_denoised.mif /Users/luis/Desktop/Projects/Advanced_visual_system/**te st**/dwipreproc-tmp-9DT0RP/dwi.mif

Cheers!

Luis

**Comment by Lestropie** [comment]
Hi Luis,

Generally better I think to make a new issue and reference any other issues or PRs that might be relevant: easier to manage than having multiple topics within a single issue that might not be caught by a search sometime down the line :+1: Worst case scenario, it gets tagged as `duplicate` or `wontfix`, no biggie.

This *should* have been dealt with in #1331, which is included in `3.0_RC3`. Are you using a version older than that?

**Comment by lmlacerda** [comment]
Hi Rob,

Copy that! Will do so in the future.

I just checked and I am still using 3.0_RC1, so that might be the issue - I will update my current installation.

Thanks for flagging this up.

Cheers

Luis
